collection-scripts/gather_ctlplane_resources

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# load shared functions and data
+DIR_NAME=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# shellcheck disable=SC1091
+source "${DIR_NAME}/common.sh"
+
+NS="$1"
+if [ -z "$NS" ]; then
+    echo "No namespace passed, using the default one"
+    NS=openstack
+fi
+
+# Only get resources if the namespace exists
+if ! check_namespace "${NS}"; then
+    exit 0
+fi
+
+declare resources=(
+    "services"
+    "routes"
+    "daemonset"
+    "replicaset"
+    "deployments"
+    "statefulset"
+    "jobs"
+)
+
```

Comment:
We can also capture the status of all the pods.
```
oc get pods > "${NAMESPACE_PATH}"/"$NS"/${NS}_pods.log
```
We are capturing it[ here](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/artifacts/tasks/cluster_info.yml#L44) in ci-framework.

Reply:
The view of the pods is captured as part of `get all`.
In my last iteration I added `get all`, which is saved in `"${NAMESPACE_PATH}"/"${NS}"/all_resources.log`.
`oc get all` basically dumps that view with a single command, a `get pods` would be redundant because we already have that info [1].
In addition, you can see we get all the `events` and the `pvc` associated with the ctlplane deployment (the regular oc must-gather tool can get additional info about `storageClasses` and `PVs`, that should be out of scope).

@raukadah is that sufficient or do you think we need to get more?

[1] https://paste.opendev.org/show/b27p4bzrlR9Z8ztfX6Qc/

Reply:
Thank you @fmount for adding that, I tested the latest image v61. and here are the logs: 

- https://logserver.rdoproject.org/59/359/4ae53455e0ca1f00282fdb25d5469e7dd19fac24/github-check/ci-framework-crc-podified-edpm-baremetal/04b056b/ci-framework-data/logs/quay-io-fpantano-must-gather-sha256-d9b2936e6e683c47ed67a34a706ad2e2f02a3d120ae71111968868a31206639b/namespaces/openstack-operators/all.log
- https://logserver.rdoproject.org/59/359/4ae53455e0ca1f00282fdb25d5469e7dd19fac24/github-check/ci-framework-crc-podified-edpm-baremetal/04b056b/ci-framework-data/logs/quay-io-fpantano-must-gather-sha256-d9b2936e6e683c47ed67a34a706ad2e2f02a3d120ae71111968868a31206639b/namespaces/openstack-operators/events.log
- https://logserver.rdoproject.org/59/359/4ae53455e0ca1f00282fdb25d5469e7dd19fac24/github-check/ci-framework-crc-podified-edpm-baremetal/04b056b/ci-framework-data/logs/quay-io-fpantano-must-gather-sha256-d9b2936e6e683c47ed67a34a706ad2e2f02a3d120ae71111968868a31206639b/namespaces/openstack-operators/pvc.log

It covers all the stuff what ci-framework needs.